src/solvers/refinement/string_constraint_generator.cpp

```diff
@@ -0,0 +1,1840 @@
+    return string_insert(expr);
+  else if(starts_with(id,cprover_string_insert_int_func))
+    return string_insert_int(expr);
+  else if(starts_with(id,cprover_string_insert_long_func))
+    return string_insert_long(expr);
+  else if(starts_with(id,cprover_string_insert_bool_func))
+    return string_insert_bool(expr);
+  else if(starts_with(id,cprover_string_insert_char_func))
+    return string_insert_char(expr);
+  else if(starts_with(id,cprover_string_insert_double_func))
+    return string_insert_double(expr);
+  else if(starts_with(id,cprover_string_insert_float_func))
+    return string_insert_float(expr);
+  else if(starts_with(id,cprover_string_insert_char_array_func))
+    return string_insert_char_array(expr);
+  else if(starts_with(id,cprover_string_substring_func))
+    return string_substring(expr);
+  else if(starts_with(id,cprover_string_trim_func))
+    return string_trim(expr);
+  else if(starts_with(id,cprover_string_to_lower_case_func))
+    return string_to_lower_case(expr);
+  else if(starts_with(id,cprover_string_to_upper_case_func))
+    return string_to_upper_case(expr);
+  else if(starts_with(id,cprover_string_char_set_func))
+    return string_char_set(expr);
+  else if(starts_with(id,cprover_string_value_of_func))
+    return string_value_of(expr);
+  else if(starts_with(id,cprover_string_empty_string_func))
+    return empty_string(expr);
+  else if(starts_with(id,cprover_string_copy_func))
+    return string_copy(expr);
+  else if(starts_with(id,cprover_string_of_int_func))
+    return of_int(expr);
+  else if(starts_with(id,cprover_string_of_int_hex_func))
+    return of_int_hex(expr);
+  else if(starts_with(id,cprover_string_of_float_func))
+    return of_float(expr);
+  else if(starts_with(id,cprover_string_of_double_func))
+    return of_double(expr);
+  else if(starts_with(id,cprover_string_of_long_func))
+    return of_long(expr);
+  else if(starts_with(id,cprover_string_of_bool_func))
+    return of_bool(expr);
+  else if(starts_with(id,cprover_string_of_char_func))
+    return of_char(expr);
+  else if(starts_with(id,cprover_string_of_char_array_func))
+    return of_char_array(expr);
+  else if(starts_with(id,cprover_string_set_length_func))
+    return string_set_length(expr);
+  else if(starts_with(id,cprover_string_delete_func))
+    return string_delete(expr);
+  else if(starts_with(id,cprover_string_delete_char_at_func))
+    return string_delete_char_at(expr);
+  else if(starts_with(id,cprover_string_replace_func))
+    return string_replace(expr);
+  else if(starts_with(id,cprover_string_data_func))
+    return string_data(expr);
+  else
+    {
+      std::string msg("string_exprt::function_application: unknown symbol :");
+      msg+=id.c_str();
+      throw msg;
+    }
+}
+
+
+irep_idt extract_java_string(const symbol_exprt & s)
+{
+  std::string tmp(s.get(ID_identifier).c_str());
+  std::string value = tmp.substr(31);
+  return irep_idt(value);
+}
```

Comment:
It will be when the better representation gets into CBMC but as far as I know, it's not yet the case.
We can either let it like that and make another pull request with the new representation and fixing this hack; or merge a pull request with the new representation before this one.